src/accelerate/utils/modeling.py

```diff
@@ -811,7 +811,7 @@ def load_state_dict(checkpoint_file, device_map=None):
         if device_map is None:
             return safe_load_file(checkpoint_file)
         else:
-            devices = [device for device in device_map.values() if device not in ["disk"]]
+            devices = list({device: () for device in device_map.values() if device not in ["disk"]})
```

Comment:
```suggestion
            devices = list(set(device_map.values()) - {"disk"})
```
Might be shorter like this then, and easier to understand.

Reply:
since sets are unordered, this will change the uploading pattern to be nondeterministic and reveal information about the memory addresses of the hashed objects. with the dict, the first weights are uploaded to the first device, but with the set a random device will be uploaded to first. is this what you would like?

Reply:
The order is not used here in this function, so this shouldn't be an issue.

Reply:
You're right, I committed your change.